src/sysupdate/sysupdated.c

```
@@ -0,0 +1,1911 @@
+        Job *j = ASSERT_PTR(userdata);
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(json_variant_unrefp) JsonVariant *json = NULL;
+        int r;
+
+        assert(s);
+        assert(si);
+
+        if (IN_SET(j->type, JOB_UPDATE, JOB_VACUUM)) {
+                assert(j->target->busy);
+                j->target->busy = false;
+        }
+
+        if (si->si_code != CLD_EXITED) {
+                assert(IN_SET(si->si_code, CLD_KILLED, CLD_DUMPED));
+                sd_bus_error_setf(&error, SD_BUS_ERROR_FAILED,
+                                  "Job terminated abnormally with signal %s.",
+                                  signal_to_string(si->si_status));
+        } else if (si->si_status != EXIT_SUCCESS)
+                if (j->status_errno)
+                        sd_bus_error_set_errno(&error, j->status_errno);
+                else
+                        sd_bus_error_setf(&error, SD_BUS_ERROR_FAILED,
+                                          "Job failed with exit code %i.", si->si_status);
+        else {
+                r = job_parse_stdout(TAKE_FD(j->stdout_fd), &json);
+                if (r < 0)
+                        sd_bus_error_set_errnof(&error, r, "Failed to parse JSON: %m");
+        }
+
+        r = sd_bus_emit_signal(
+                        j->manager->bus,
+                        "/org/freedesktop/sysupdate1",
+                        "org.freedesktop.sysupdate1.Manager",
+                        "JobRemoved",
+                        "uoi",
+                        j->id,
+                        j->object_path,
+                        j->status_errno ?: si->si_status);
+        if (r < 0)
+                log_warning_errno(r, "Cannot emit JobRemoved message, ignoring: %m");
+
+        if (j->dbus_msg && j->complete_cb) {
+                if (sd_bus_error_is_set(&error))
+                        sd_bus_reply_method_error(j->dbus_msg, &error);
```

it's ok to ignore errors sometimes in particular if they are in a cascade like here, but please at least log about this, i.e. add `if (r < 0) log_warning_errno(r, "…, ignoring: %m");` here, just like you did above